api/pom.xml

```diff
@@ -96,6 +96,11 @@
       </exclusions>
     </dependency>
     <dependency>
+      <groupId>org.apache.hadoop</groupId>
```

Comment:
It's for ApplicationId that the users can use for their own purposes and I am also using in shutdown of the app. I guess it would be possible to return a string applicationId but I think now or in future we may want to include other information from Yarn or even other types of launchers that would be useful to the user.

I didnt understand the secomd comment.